services/chaindb/postgresql/validatorepochsummaries.go

```diff
@@ -533,15 +533,27 @@ func (s *Service) PruneValidatorEpochSummaries(ctx context.Context, to phase0.Ep
 
 	queryBuilder.WriteString(`
 DELETE FROM t_validator_epoch_summaries
-WHERE f_epoch <= $1
+USING t_validators
+WHERE t_validator_epoch_summaries.f_validator_index = t_validators.f_index
+AND f_epoch <= $1
 `)
 	queryVals = append(queryVals, to)
 
-	if len(retain) > 0 {
+	if len(retainPubkeys) > 0 {
 		queryBuilder.WriteString(`
-AND  f_validator_index NOT IN($2)
+AND NOT (t_validators.f_public_key = ANY($2))
 `)
-		queryVals = append(queryVals, retain)
+
+		retainPubkeysBytes := make([][]byte, 0, len(retainPubkeys))
+
+		for _, pubkey := range retainPubkeys {
+			pubkeyBytes := make([]byte, len(pubkey))
+			copy(pubkeyBytes, pubkey[:])
```

Comment:
Is this required?  Could it be replaced by just `pubKey[:]` to reduce allocations and copying?

Reply:
Replacing by `pubKey[:]` would work with 1.22 go for loops.
It does not work with 1.21 go (project use 1.20 go, and i did not want to change version)

See example here: https://go.dev/play/p/K5u3iVSeWe_e

Reply:
I think you should be able to avoid that by only using the loop variable:

```
	for i := range retain {
		pubkeysBytes = append(pubkeysBytes, retain[i][:])
	}
```

Reply:
Oh, that's better ! Thanks.